packages/react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -355,16 +349,18 @@ class RCTAztecView: Aztec.TextView {
     }
 
     // MARK: - Dictation
-
-    override func dictationRecordingDidEnd() {
-        isInsertingDictationResult = true
-    }
-
-    public override func insertDictationResult(_ dictationResult: [UIDictationPhrase]) {
-        let objectPlaceholder = "\u{FFFC}"
-        let dictationText = dictationResult.reduce("") { $0 + $1.text }
-        isInsertingDictationResult = false
-        self.text = self.text?.replacingOccurrences(of: objectPlaceholder, with: dictationText)
+    
+    func removeUnicodeAndRestoreCursor(from textView: UITextView) {
+        // Capture current cursor position
+        let originalPosition = textView.offset(from: textView.beginningOfDocument, to: textView.selectedTextRange?.start ?? textView.beginningOfDocument)
+                
+        // Replace occurrences of the obj symbol ("\u{FFFC}")
+        textView.text = textView.text?.replacingOccurrences(of: "\u{FFFC}", with: "")
+                
+        if let newPosition = textView.position(from: textView.beginningOfDocument, offset: originalPosition) {
```

Comment:
Without this code, the cursor would always be repositioned (along with the dictated text) after dictating in the middle of a paragraph. I experimented with a few different ways to repositioning the cursor, and this is the one that seemed to work the best.

I've been able to reproduce one issue with this approach:

1. Begin dictating text into a paragraph block.
2. _Without ending dictation_, move the cursor into the middle of the text and begin dictating.
3. Notice that the cursor can appear out of place when the text moves onto a new line.
4. The cursor will correct itself as you continue dictating, and the text will remain in the correct position.

Note, this _doesn't_ happen if you end dictation and then re-start it to add text into the middle of the text. I'd consider this somewhat of an edge case, which corrects itself and doesn't impact the actual text entry.

In the interest of shipping a timely fix, and if there aren't any obvious other solutions, I'd like to propose that we go with this one for now and address improving the cursor positioning in a separate PR.

Reply:
That sounds reasonable to me @SiobhyB. 👍